gpflow/models/training_mixins.py

```diff
@@ -0,0 +1,58 @@
+import abc
+import collections
+from typing import Callable, Optional, Tuple, TypeVar, Union
+
+
+import tensorflow as tf
+import numpy as np
+
+InputData = tf.Tensor
+OutputData = tf.Tensor
+RegressionData = Tuple[InputData, OutputData]
```

Comment:
is `RegressionData` still accurate if we're doing GPC?

Reply:
It was already called RegressionData before, it just got moved from model.py to trainin_mixins.py so the Mixin classes have access to it. I agree it's not quite perfect; the key distinction here is between models that take both X and Y and models that only take X (point processes) or Y (latent variable models). If you can come up with a better name, please make a suggestion.